README.md

```diff
@@ -122,6 +122,172 @@ your container. This will allow you to login to the root account remotely. Local
+* `/usr/share/container-layer/mysql/post-init/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before the first start of container (i.e. in case the container is started with empty data directory). At the time these scripts are executed, the daemon runs on localhost only and `mysql` utility should use arguments stored in `$mysql_flags` environment variable. This directory usually includes scripts for creating new databases, users or initialize the database according to your needs.
+* `/usr/share/container-layer/mysql/pre-start/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed before every start of container (no matter whether the data directory is empty or not). At the time these scripts are executed, the daemon runs on localhost only and `mysql` utility should use arguments stored in `$mysql_flags` environment variable. Example of script in this directory is a script that allows to change password.
+* `/usr/share/container-layer/mysql/include/` -- this directory includes shell scripts with extention `.sh` that will be sourced and thus executed in the beginning of the commands like run-mysqld. Files in this directory contain usually functions or variables definition that are supposed to be used in other scripts used in the image.
+* `/usr/share/container-layer/mysql/usage/` -- this directory includes text files with extention `.txt` that will be printed as usage message for the container in case wrong arguments are given. Shell variables in these files will be expanded using `envsubst` utility.
+
+Generally, to perform some additional action according to your needs, put appropriate files into these directories in your Dockerfile. See other files in those directories for inspiration.
+
+### Example of extending the Docker image by creating layered Dockerfile
+
+This example adds posibility to add a new user during database inicialization. The username and password is specified using `MYSQL_CONTENT_ADMIN_USER` and `MYSQL_CONTENT_ADMIN_PASSWORD` environment variables, provided when starting the container the first time.
+
+Since you might need to validate the input provided by user in environment variables, it is possible to do it by droping a shell script appropriate directory, that is sourced in the beginning of the daemon start.
+
+Finally, we want to extend the usage message that is printed in case wrong arguments are given by user. This is done by droping a text file into appropriate directory
+
+Content of the layered Dockerfile:
+```
+FROM <this-image>
+USER 0
+ADD 60_content_admin_create.sh /usr/share/container-layer/mysql/post-init/
+ADD 60_content_admin_password.sh /usr/share/container-layer/mysql/pre-start/
+ADD 60_verify_content_admin.sh /usr/share/container-layer/mysql/pre-init/
+ADD 60_usage_content_admin.txt /usr/share/container-layer/mysql/usage/
```

Comment:
@hhorak that is what the layering grants you. I can imagine an use case where somebody builds an mysql image that will add some performance optimizations on top of the original mysql image (layer #1) and then publish it on Dockerhub. Other user find this image and add more configuration to it (like create different users/etc), which is layer #2. Now you have couple problems:

1) How the second user know what modifications were done in layer #1? He has to go pull the image and inspect the configuration files manually
2) What happens when the user supplying layer #1 change something that will break layer #2? I guess the user again has to go investigate
3) The second user will have to rebuilt the image everytime 1) layer #1 is updated, 2) base mysql image is updated. 

I'm just trying to highlight possible problems users might have by extending the image by adding more customization scripts into it.